website/server/controllers/api-v3/auth.js

```diff
@@ -101,23 +101,48 @@ api.registerLocal = {
     let existingUser = res.locals.user; // If adding local auth to social user
 
     req.checkBody({
-      email: {
-        notEmpty: {errorMessage: res.t('missingEmail')},
-        isEmail: {errorMessage: res.t('notAnEmail')},
-      },
       username: {
-        notEmpty: {errorMessage: res.t('missingUsername')},
-        isLength: {options: {min: USERNAME_LENGTH_MIN, max: USERNAME_LENGTH_MAX}, errorMessage: res.t('usernameWrongLength')},
+        notEmpty: true,
+        errorMessage: res.t('missingUsername'),
         // TODO use the constants in the error message above
+        isLength: {options: {min: USERNAME_LENGTH_MIN, max: USERNAME_LENGTH_MAX}, errorMessage: res.t('usernameWrongLength')},
         matches: {options: /^[-_a-zA-Z0-9]+$/, errorMessage: res.t('usernameBadCharacters')},
       },
+      email: {
+        notEmpty: true,
+        errorMessage: res.t('missingEmail'),
+        isEmail: {errorMessage: res.t('notAnEmail')},
+      },
       password: {
-        notEmpty: {errorMessage: res.t('missingPassword')},
+        notEmpty: true,
+        errorMessage: res.t('missingPassword'),
         equals: {options: [req.body.confirmPassword], errorMessage: res.t('passwordConfirmationMatch')},
       },
     });
+
     let validationErrors = req.validationErrors();
-    if (validationErrors) throw validationErrors;
+    if (validationErrors) {
+      let errors = [];
+      let usernameCheck = false;
+      let emailCheck = false;
+      let passwordCheck = false;
+      // only show the first error of each param
+      for (let e of validationErrors) {
+        if (!usernameCheck && e.param === 'username') {
+          errors.push(e);
+          usernameCheck = true;
+        }
+        if (!emailCheck && e.param === 'email') {
+          errors.push(e);
+          emailCheck = true;
+        }
+        if (!passwordCheck && e.param === 'password') {
+          errors.push(e);
+          passwordCheck = true;
+        }
+      }
+      throw errors;
+    }
```

Comment:
Not sure if this is the most efficient way to do this, so any suggestions are welcome!